source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp

```diff
@@ -130,26 +171,13 @@ ShaderPtr GlslFragmentGenerator::generate(const string& fragmentName, ElementPtr
     const bool lighting = graph.hasClassification(ShaderNode::Classification::SHADER | ShaderNode::Classification::SURFACE)
         || graph.hasClassification(ShaderNode::Classification::BSDF);
 
-    // Emit lighting functions
-    if (lighting)
-    {
-        const VariableBlock& lightData = pixelStage.getUniformBlock(HW::LIGHT_DATA);
-        emitLine("struct " + lightData.getName(), pixelStage, false);
-        emitScopeBegin(pixelStage);
-        emitVariableDeclarations(lightData, EMPTY_STRING, Syntax::SEMICOLON, context, pixelStage, false);
-        emitScopeEnd(pixelStage, true);
-        emitLineBreak(pixelStage);
-        emitLine("uniform " + lightData.getName() + " " + lightData.getInstance() + "[MAX_LIGHT_SOURCES]", pixelStage);
-        emitLineBreak(pixelStage);
-    }
-
     // Emit common math functions
     emitInclude("pbrlib/genglsl/lib/mx_math.glsl", context, pixelStage);
     emitLineBreak(pixelStage);
 
     if (lighting)
     {
-        emitSpecularEnvironment(context, pixelStage);
+        emitBlock(MX_ENVIRONMENT_MAYA, context, pixelStage);
```

Comment:
Check with @niklasharrysson , which was Jerry's question. Are we okay with adding a MAYA specific block here ?
I don't think anyone else is using this generator but double-checking. the ogsxml fragement generator for sure is Maya centric.

Reply:
I don't mind adding Maya specific code here. This generator was targeting Maya from it's origin and there are probably other Maya specific constructs in here :)

Reply:
Okay. Just checking.